### PR TITLE
Bring in ERB-lint, refactor view code, fix indentation, etc

### PR DESCRIPTION
**Pull request by murny**
Builds on top of #1485 

Part of #1280. 

Refactor most of the HTML in the views with the help of ERB-lint

### CODE REVIEW

File: .erb-lint.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    # enforced_style: 'never'
+  SpaceAroundErbTag:
+    enabled: true
+  SpaceInHtmlTag:
+    enabled: true
+  SpaceIndentation:
+    enabled: true
+  TrailingWhitespace:
+    enabled: true
+  # TODO: Fix this, we shouldn't be using hard coded strings in ERB, use i18n instead
```
Comment (murny) [inline .erb-lint.yml]: Disabled i18n for now, but we should circle back and fix this

File: app/views/catalog/_facet_pagination.html.erb [hunk 2]
```diff
@@ -1,21 +1,39 @@
-    <div class="prev_next_links btn-group pull-left">
-      <%= link_to_previous_page @pagination, raw(t('views.pagination.previous')), :params => params, :param_name => Blacklight::Solr::FacetPaginator.request_keys[:page], :class => 'btn btn-link', :data => {:ajax_modal => "preserve"} do %>
-      <span class="disabled btn btn-disabled"><%= raw(t('views.pagination.previous')) %></span>
-    <% end %>
-      <%= link_to_next_page @pagination, raw(t('views.pagination.next')), :params => params, :param_name => Blacklight::Solr::FacetPaginator.request_keys[:page], :class => 'btn btn-link',  :data => {:ajax_modal => "preserve"} do %>
+<div class="prev_next_links btn-group pull-left">
+  <%= link_to_previous_page @pagination,
```
Comment (murny) [inline app/views/catalog/_facet_pagination.html.erb]: Try to keep the line length for code/html/etc to be under 120 characters. 

For things like this `link_to` which has lots of options, you can add a newline on each new option making it far more readable

File: app/views/catalog/_index_default.html.erb [hunk 3]
```diff
@@ -1,59 +1,56 @@
...
-      <!-- Display format note -->
-      <% if document.to_h["format"] and document.to_h["electronic_tesim"] %>
-        <% if document.to_h["format"].first == "Book" and document.to_h["electronic_tesim"].include? "Online" %>
-          <dt class="blacklight-ebook">Note:</dt> <dd class="blacklight-ebook">This is an <strong>eBook</strong>.</dd>
-        <% elsif document.to_h["format"].first == "Journal" and document.to_h["electronic_tesim"].include? "Online" %>
-          <dt class="blacklight-ebook">Note:</dt> <dd class="blacklight-ebook">This is an <strong>Electronic Journal</strong>.</dd>
-        <% elsif document.to_h["format"].first == "Journal" and document.to_h["electronic_tesim"].include? "At Library" %>
-          <dt class="blacklight-ebook">Note:</dt> <dd class="blacklight-ebook">This is a <strong>Print Journal</strong>.</dd>
-        <% end %>
+      <%# TODO: Closing </dt> below is useless? Also why are the following closing tags outside of the if statement?
```
Comment (murny) [inline app/views/catalog/_index_default.html.erb]: Got no idea whats happening here? We got these two closing tags...the </dt> doesnt have a corresponding opening tag...and the </dd i assume corresponds to the opening tag up on Line 18 but this is all surrounded by if statements...so if those conditions evaluate to false...then we have a extra closing tag floating around...

File: app/views/catalog/_kule_holdings.html.erb [hunk 4]
```diff
@@ -1,12 +1,11 @@
-
```
Comment (murny) [inline app/views/catalog/_kule_holdings.html.erb]: Good mini example of how bad the white space/indentation is on these partials (extra new line at top, everything is floated over by 6-10 spaces for some reason etc?)

File: app/views/catalog/_facets.html.erb [hunk 5]
```diff
@@ -1,22 +1,21 @@
-<% # main container for facets/limits menu -%>
```
[the remark this answers is not in the document]
Comment (pgwillia) [reply app/views/catalog/_facets.html.erb]: 🤦

File: Gemfile [hunk 6]
```diff
@@ -59,6 +59,8 @@ group :development do
   gem 'better_errors', '>= 2.3.0'
   gem 'binding_of_caller'
 
+  gem 'erb_lint', '0.0.28', require: false
```
Comment (pgwillia) [inline Gemfile]: Will make it more difficult to test Ruby 2.1.5 in development. Really need https://github.com/ualbertalib/discovery/issues/1432

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,9 @@ and releases in Discovery project adheres to [Semantic Versioning](http://semver
 
 ## [Unreleased]
 
+### Added
+- Bring in ERB-lint, refactor view code, fix identations, etc [#1486](https://github.com/ualbertalib/discovery/pull/1486)
```
Comment (pgwillia) [inline CHANGELOG.md]: :+1:

File: Gemfile (hunk 6 above)
Comment (pgwillia) [reply Gemfile]: Just another underline to getting https://github.com/ualbertalib/discovery/issues/1432 done

**Pull request merged**